server/src/main/java/org/elasticsearch/common/collect/ImmutableOpenMap.java

```diff
@@ -136,6 +138,24 @@ public void remove() {
         };
     }
 
+    /**
+     * Returns a {@link Set} view of the keys contained in this map.
+     */
+    public Set<KType> keySet() {
+        final Iterator<KType> keysIterator = keysIt();
+        return new AbstractSet<>() {
+            @Override
+            public Iterator<KType> iterator() {
+                return keysIterator;
```

Comment:
Shouldn't iterator() create a new iterator, ie call keysIt()?

Reply:
`keysIterator` is already holds the new iterator from `keysIt()`. But I can also directly call `keysIt` here

Reply:
Done!